### ISSUE TITLE
validations.errors not updating when hasMany validation changes

### ISSUE THREAD

**Issue by jameshoward** [offirgolan/ember-cp-validations#322]
### Environment

- Ember Version: 2.6.0
- Ember CLI Version: 2.4.2
- Ember CP Validations Version: 2.9.4

### Steps to Reproduce

* Add has many validation to a model with a has many to an object that also has more than one validator defined. E.g. team has many players, player has a `firstName` and `lastName` that must both be present.
* Add a new empty player. Validation (`team.validation.errors`) correctly includes that `firstName` and `lastName` must both be present.
* Set `firstName` on the player.

Expected: `team.validation.errors` no longer contains an entry for the `firstName` field, just the `lastName`.
Actual: `team.validation.errors` includes the original two errors.

If team has further validation rules (e.g. it's `teamName` must be set) then the `team.validations.errors` correctly updates if the parent validation state changes. It's just that if you change the child model but not to the point that `isValid`/`isTruelyValid` is true, the parent errors don't update. Worth noting the child ones do (e.g. `player.validations.errors`).

Not sure if this is an issue or something I'm doing wrong / expecting that isn't right.

**Comment by offirgolan** [comment]
@jameshoward can you post some snippets for your models / validations?

**Comment by jameshoward** [comment]
I've built a test project that demonstrates the issues I'm having: https://github.com/jameshoward/ember-cp-validations-test

You can either dig around in the code or fire it up and have a look. Here's what I'm seeing:

1. Initial state shows that the team is invalid because `name`is blank. Correct.
2. Add a player. Validation for the player is correct as `firstName` / `lastName` is blank. However, the validation errors for the team do not include the hasMany errors. (This may be correct behaviour but isn't consistent…)
3. Enter a `name` for the team. The team validation errors update to show the issues with the player. This seems correct, but it's odd they only appear at this point.
4. Enter a `firstName` for the player. The player's validation correctly updates but the team validation suggests the `firstName` is still blank.
5. Enter a `lastName` for the player. All validation is now fine.

Things get even more funky if you add 2 players then start entering values. At that point I can never see any of the hasMany validations showing for the team.

If the idea is that `validations.errors` doesn't include `hasMany` errors, that's cool and I can work around that but then no. 3, where they do show up, doesn't make sense.

Any help greatly appreciated. This module is really helping with the validation and this has been the only thing I'm struggling with a bit.

**Comment by offirgolan** [comment]
@jameshoward thanks for putting this together. Ill take a closer look when I have some time later this week.

**Comment by offirgolan** [comment]
@jameshoward I've been working on a pretty big internal re-work the last couple of days which just landed in master this evening (#321). When I ran your dummy app and used the current version in master, it seemed to solve the issue! 🎉 

Im waiting on another PR to land and then ill release another beta version with this fix. Thanks for taking the time to put the dummy app together 😸

**Comment by jameshoward** [comment]
Excellent, sounds good. No worries on the dummy app, happy to help since this library is doing great stuff for my project.

**Comment by jameshoward** [comment]
Just FYI, I've just updated our app to use v3.0.0 and it's working as expected now. Thanks again!

**Comment by offirgolan** [comment]
@jameshoward thats great to hear! Thanks for the confirmation! 😺
